src/ExpenseTracker.Web/Middleware/WorkspaceMiddleware.cs

```diff
@@ -11,36 +12,38 @@ public class WorkspaceMiddleware
     {
         private const string WorkspaceCreateUrl = "/Workspace/Create";
         private const string LoginUrl = "/Account/Login";
-        
+
 
         private static readonly List<string> PathsToAvoid = new()
         {
             LoginUrl,
             WorkspaceCreateUrl
         };
-        
+
         private readonly RequestDelegate _next;
+
         public WorkspaceMiddleware(RequestDelegate next)
         {
             _next = next;
         }
+
         public async Task Invoke(HttpContext httpContext, IUserProvider userProvider)
         {
             var currentRequestPath = httpContext.Request.Path;
 
             var currentUser = await userProvider.GetCurrentUser();
```

Comment:
@BirajMainali This bit is not very performant. We could add some optimizations here.

1. Just retrieve the userId instead of the user entity
2. Use Workspace repo to check if the userId has default and active workspace

This should make the code more readable and performant as well.